Firestore/Source/Core/FSTView.mm

```diff
@@ -433,20 +435,22 @@ - (BOOL)shouldBeLimboDocumentKey:(const DocumentKey &)key {
 /**
  * Updates syncedDocuments and current based on the given change.
  */
-- (void)applyTargetChange:(nullable FSTTargetChange *)targetChange {
-  if (targetChange) {
-    for (const DocumentKey &key : targetChange.addedDocuments) {
+- (void)applyTargetChange:(const absl::optional<TargetChange> &)maybeTargetChange {
+  if (maybeTargetChange.has_value()) {
```

Comment:
This is fine (and is exactly as previous). You could reverse the if statement to eliminate a level of indentation, i.e.:

```
if (!...) return;
...
```

Very optional.

Reply:
I would also prefer an early return, but other platforms follow this pattern as well. I'd rather leave as is for consistency.